src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,68 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instaces = new HashMap<>();
+    private final Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
+
+    static {
+        injector.interfaceImplementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        injector.interfaceImplementations.put(ProductParser.class, ProductParserImpl.class);
+        injector.interfaceImplementations.put(ProductService.class, ProductServiceImpl.class);
+    }
```

Comment:
check common mistakes, we can do it via a `Map.of(key, value, key, value, ...)` instead
```suggestion
    private final Map<Class<?>, Class<?>> interfaceImplementations = Map.of(...);
```

Reply:
yeah!